### DISCUSSION TITLE
Consider moving Imenu support into `magit-section`

### DISCUSSION THREAD

**Discussion by alphapapa** [magit/magit#5104] [Ideas]
Hi @tarsius,

As you mentioned at https://github.com/magit/magit/pull/4999#issuecomment-1703950636, I'm pinging you to ask about moving Imenu-related code into `magit-section` so that all `magit-section`-using libraries could benefit from it.  I'd like to have Imenu support in my `listen-library` buffers, which use `magit-section`, but obviously I don't want to require `magit` to do so, and it'd be preferable to not copy the code, since it should be applicable to all `magit-section`-derived buffers, ISTM.

Thanks.

**Comment by tarsius** [comment]
Huhu Adam,

Done in 76c14efc787306a9d2d45a74a8e6754db89c4813.

> ISTM

International Society of Travel Medicine?
Institute of Secretariat Training and Management?
Institut für Strömungsmechanik?

**Comment by alphapapa** [comment]
@tarsius Well, thank you, but... I was asking about Imenu support, not bookmark support.  

I'm not sure how feasible it would be to provide generic bookmark support for any `magit-section-mode`-derived mode.  Thinking of all of mine, there are various buffer-local variables and arguments that would need to be provided to the function that sets up the buffer, which I provide in my own bookmark make-record and handler functions.

> ISTM

It Seems To Me

**Reply by tarsius** [reply]
> I was asking about Imenu support, not bookmark support.

Ups. I've moved Imenu support too now. Not a complete loss that I move bookmark support first; seems to me it makes sense to make those two available at the same level.

> I'm not sure how feasible it would be to provide generic bookmark support for any `magit-section-mode`-derived mode.  Thinking of all of mine, there are various buffer-local variables and arguments that would need to be provided to the function that sets up the buffer, which I provide in my own bookmark make-record and handler functions.

Might not work for you, but looking at how this is used in `magit-bookmarks.el` I think it could be enough in some third-party cases.

**Reply by alphapapa** [reply]
That's great!  Thank you very much.  This will be very helpful in my packages!

**Reply by tarsius** [reply]
You are welcome!
